From 7d5bda9975bff4bc1ecf9260af868b2754522acc Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Mon, 11 Jun 2018 10:33:24 +0800
Subject: [PATCH 3947/5242] MLK-18552-1 ARM64: dts: add clock disp_axi and
 disp_apb for CSI and MIPI CSI

commit  eb171b021d99335e376f275d29ba827455e2867a from
https://source.codeaurora.org/external/imx/linux-imx.git

Add nodes for clock disp_axi and disp_apb for access to avoid hang issue.
On i.MX8MM, use the register in CSI to do MIPI PHY reset, so add these
clocks for MIPI CSI driver as well.

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Reviewed-by: Guoniu.Zhou <guoniu.zhou@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi |   10 ++++++----
 1 file changed, 6 insertions(+), 4 deletions(-)

diff --git a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
index e0168c6..9fa76b9 100755
--- a/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-imx8mm.dtsi
@@ -170,9 +170,9 @@
 		compatible = "fsl,imx8mm-csi", "fsl,imx8mq-csi", "fsl,imx6s-csi";
 		reg = <0x0 0x32e20000 0x0 0x10000>;
 		interrupts = <GIC_SPI 16 IRQ_TYPE_LEVEL_HIGH>;
-		clocks = <&clk IMX8MM_CLK_DUMMY>,
+		clocks = <&clk IMX8MM_CLK_DISP_AXI_ROOT>,
 			<&clk IMX8MM_CLK_CSI1_ROOT>,
-			<&clk IMX8MM_CLK_DUMMY>;
+			<&clk IMX8MM_CLK_DISP_APB_ROOT>;
 		clock-names = "disp-axi", "csi_mclk", "disp_dcic";
 		status = "disabled";
 	};
@@ -183,8 +183,10 @@
 		interrupts = <GIC_SPI 17 IRQ_TYPE_LEVEL_HIGH>;
 		clock-frequency = <333000000>;
 		clocks = <&clk IMX8MM_CLK_CSI1_CORE_DIV>,
-				<&clk IMX8MM_CLK_CSI1_PHY_REF_DIV>;
-		clock-names = "mipi_clk", "phy_clk";
+			<&clk IMX8MM_CLK_CSI1_PHY_REF_DIV>,
+			<&clk IMX8MM_CLK_DISP_AXI_ROOT>,
+			<&clk IMX8MM_CLK_DISP_APB_ROOT>;
+		clock-names = "mipi_clk", "phy_clk", "disp_axi", "disp_apb";
 		bus-width = <4>;
 		status = "disabled";
 	};
-- 
1.7.9.5

